assets/typst/client/game_ui.typ

```diff
@@ -192,96 +206,126 @@
 
 // Rule
 // - Each box represents 10 HP
-#let playerhealth(current_hp, max_hp, rect_width: 21pt, rect_height: 31pt, spacing: 8pt) = {
-    // Calculate the ratio of current HP to max HP
-    let hp_ratio = current_hp / max_hp
+#let playerhealth(
+  current_hp,
+  max_hp,
+  total_width: 280pt,
+  rect_height: 31pt,
+  spacing: 8pt,
+) = {
+  // Calculate the total number of blocks needed
+  let blocks = calc.ceil(current_hp / 10)
+  let max_blocks = calc.ceil(max_hp / 10)
 
-    // Determine the total number of blocks and how many blocks to display based on HP ratio
-    let max_blocks = 10  // Adjust this value for more or fewer blocks
-    let num_blocks = hp_ratio * max_blocks
-    
-    // Define health state colors
-    let healthy_color = rgb("A9DC76")    // Green
-    let medium_color = rgb("FFB454")     // Orange
-    let low_color = rgb("FF6188")        // Red
+  // Calculate the width of each block
+  let rect_width = (total_width - (max_blocks - 1) * spacing) / max_blocks
 
-    // Function to determine block color based on health ratio
-    let get_block_color(is_filled) = {
-        if not is_filled {
-            // For empty blocks, use transparent version of healthy color
-            return healthy_color.transparentize(80%)
-        }
+  // Define health state colors
+  let low_color = rgb("FF6188") // Red
+  let medium_color = rgb("FFB454") // Orange
+  let medium_high_color = rgb("FFEB3B") // Bright Yellow
+  let healthy_color = rgb("A9DC76") // Green
```

Comment:
Do you wanna use monokai pro color scheme? (simply use `red`, `orange`, `green` will do, u can also mix color if you want: `yellow.mix(base7)` or `yellow.mix((base7, 50%))`)

Reply:
Tho rmb to `import monokai_pro.typ: *` first I think.